docs/plots/index.rst

```diff
@@ -54,16 +54,26 @@ Example
 Here is the data we wish to display: a set of consecutive
 bins, defining the edges of the bins, and the counts in each bin:
 
-    >>> import numpy as np
-    >>> edges = np.asarray([-10, -5, 5, 12, 17, 20, 30, 56, 60])
-    >>> y = np.asarray([28, 62, 17, 4, 2, 4, 125, 55])
+.. plot::
+   :context:
+   :nofigs:
+   :include-source:
+
+   >>> import numpy as np
+   >>> edges = np.asarray([-10, -5, 5, 12, 17, 20, 30, 56, 60])
+   >>> y = np.asarray([28, 62, 17, 4, 2, 4, 125, 55])
```

Comment:
I am too lazy to go hunting in the sphinx documentation to find out why we need to mark up these code segments that are not generating plots. I assume it's because they are doing stuff (e.g. importing modules, setting up variables) that we need in late plot calls.

What I don't understand is why there's a plot directive for the example below which says

```
.. plot::
   :context:
   :nofigs:
   :include-source:

   >>> print(dplot.plot_prefs)
   {'xerrorbars': False, 'yerrorbars': True, 'ecolor': None, 'capsize': None, 'barsabove': False, 'xlog': False, 'ylog': False, 'linestyle': 'None', 'linecolor': None, 'color': None, 'marker': '.', 'markerfacecolor': None, 'markersize': None, 'xaxis': False, 'ratioline': False}
```

As far as I can see, we don't need to include this for plot purposes. Is it perhaps linked to the need to also run these through doctestplus?

Reply:
I am not 100% sure, so I will check again, but I think I did that to make it run in the same context where dplot is defined. My understanding is (but the might be wrong) that the code is split: Code under the plot directive is run to make the plots, and the other code is run separately by doctestplus. That other code would not know about `dplot`, because that's defined inside the plot directive code.

Reply:
That sounds logical.